cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -426,9 +451,62 @@ public Set<ProgramId> filterProgramsExistence(Collection<ProgramId> programIds)
   }
 
 
-  public void writeApplication(String namespaceId, String appId, String versionId, ApplicationSpecification spec)
-    throws IOException {
-    writeApplicationSerialized(namespaceId, appId, versionId, GSON.toJson(new ApplicationMeta(appId, spec)));
+  public void writeApplication(String namespaceId, String appId, String versionId, ApplicationSpecification spec,
+                               ChangeDetail change) throws IOException {
+    writeApplicationSerialized(namespaceId, appId, versionId, GSON.toJson(new ApplicationMeta(appId, spec, null)),
+                               change.getCreationTimeMillis(), change.getAuthor(), change.getDescription());
+  }
+
+  /**
+   * Filter the given set of programs and return those that exist.
```

Comment:
was this copied from elsewhere?

Reply:
oh yes, this was copied by mistake. I'll fix it thanks!